include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_SRC_ALPHA,
+    SDL_GPU_BLENDFACTOR_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_SRC_ALPHA_SATURATE
+} SDL_GpuBlendFactor;
+
+typedef enum SDL_GpuColorComponentFlagBits
+{
+    SDL_GPU_COLORCOMPONENT_R_BIT = 0x00000001,
+    SDL_GPU_COLORCOMPONENT_G_BIT = 0x00000002,
+    SDL_GPU_COLORCOMPONENT_B_BIT = 0x00000004,
+    SDL_GPU_COLORCOMPONENT_A_BIT = 0x00000008
+} SDL_GpuColorComponentFlagBits;
+
+typedef Uint32 SDL_GpuColorComponentFlags;
+
+typedef enum SDL_GpuFilter
+{
+    SDL_GPU_FILTER_NEAREST,
+    SDL_GPU_FILTER_LINEAR
+} SDL_GpuFilter;
+
+typedef enum SDL_GpuSamplerMipmapMode
+{
+    SDL_GPU_SAMPLERMIPMAPMODE_NEAREST,
+    SDL_GPU_SAMPLERMIPMAPMODE_LINEAR
+} SDL_GpuSamplerMipmapMode;
+
+typedef enum SDL_GpuSamplerAddressMode
+{
+    SDL_GPU_SAMPLERADDRESSMODE_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_MIRRORED_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_CLAMP_TO_EDGE
+} SDL_GpuSamplerAddressMode;
+
+/*
+ * VSYNC:
+ *   Waits for vblank before presenting.
+ *   If there is a pending image to present, the new image is enqueued for presentation.
+ *   Disallows tearing at the cost of visual latency.
+ *   When using this present mode, AcquireSwapchainTexture will block if too many frames are in flight.
+ * IMMEDIATE:
+ *   Immediately presents.
+ *   Lowest latency option, but tearing may occur.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ * MAILBOX:
+ *   Waits for vblank before presenting. No tearing is possible.
+ *   If there is a pending image to present, the pending image is replaced by the new image.
+ *   Similar to VSYNC, but with reduced visual latency.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ */
+typedef enum SDL_GpuPresentMode
+{
+    SDL_GPU_PRESENTMODE_VSYNC,
+    SDL_GPU_PRESENTMODE_IMMEDIATE,
+    SDL_GPU_PRESENTMODE_MAILBOX
+} SDL_GpuPresentMode;
+
+/*
+ * SDR:
+ *   B8G8R8A8 or R8G8B8A8 swapchain. Pixel values are in nonlinear sRGB encoding. Blends raw pixel values.
+ * SDR_LINEAR:
+ *   B8G8R8A8_SRGB or R8G8B8A8_SRGB swapchain. Pixel values are in nonlinear sRGB encoding. Blends in linear space.
+ * HDR_EXTENDED_LINEAR:
+ *   R16G16B16A16_SFLOAT swapchain. Pixel values are in extended linear encoding. Blends in linear space.
+ * HDR10_ST2048:
+ *   A2R10G10B10 or A2B10G10R10 swapchain. Pixel values are in PQ ST2048 encoding. Blends raw pixel values. (TODO: verify this)
+ */
+typedef enum SDL_GpuSwapchainComposition
+{
+    SDL_GPU_SWAPCHAINCOMPOSITION_SDR,
+    SDL_GPU_SWAPCHAINCOMPOSITION_SDR_LINEAR,
+    SDL_GPU_SWAPCHAINCOMPOSITION_HDR_EXTENDED_LINEAR,
+    SDL_GPU_SWAPCHAINCOMPOSITION_HDR10_ST2048
+} SDL_GpuSwapchainComposition;
+
+typedef enum SDL_GpuDriver
+{
+    SDL_GPU_DRIVER_INVALID = -1,
+    SDL_GPU_DRIVER_SECRET,
```

Comment:
??? Maybe a comment explaining this? :)

Reply:
Yep, will eventually have a comment similar to the ShaderFormat, which does denote that it's for NDA platforms!